h/models.py

```diff
@@ -115,13 +115,25 @@ def __acl__(self):
     }
     __analysis__ = {
         'filter': {
-            'uri': {
+            'uri_index': {
                 'type': 'pattern_capture',
                 'preserve_original': '1',
                 'patterns': [
                     '([^\\/\\?\\#\\.]+)',
                     '([a-zA-Z0-9]+)(?:\\.([a-zA-Z0-9]+))*',
                     '([a-zA-Z0-9-]+)(?:\\.([a-zA-Z0-9-]+))*',
+                    '://(.+)',
+                    '://(.+)\\?',
+                    '://(.+)\\#'
+                ]
+            },
+            'uri_search': {
+                'type': 'pattern_capture',
+                'preserve_original': '1',
+                'patterns': [
+                    '://(.+)',
+                    '://(.+)\\?',
```

Comment:
Doesn't our various selectors are for making this harder to happen? (I mean we have a selector which saves the text before and after the selected part)

But yes, that can happen, we question is, which is the least problematic.

About canonical-uri: they are not in our control, we cannot force any page owner to use them.